janome/progress.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2015 moco_beta
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+
+class ProgressHandler(ABC):
+    """
+    Base progress handler class
+    """
+
+    @abstractmethod
+    def on_start(self, total, desc=None):
+        pass
+
+    @abstractmethod
+    def on_progress(self, value=1):
+        pass
+
+    @abstractmethod
+    def on_complete(self):
+        pass
+
+
+class SimpleProgressIndicator(ProgressHandler):
+    def __init__(self, update_frequency=0.0001, format=None):
+        self.update_frequency = update_frequency
+        self.format = format or "\r{}: {:.1f}% | {}/{}"
+        self.total = None
+        self.value = None
+        self.desc = None
+
+    def print_progress(self, end=""):
+        print(self.format.format(
```

Comment:
Can you use logging.Logger instead of print? For testing, I think you can verify the output by `assertLogs()` https://docs.python.org/3.4/library/unittest.html#unittest.TestCase.assertLogs

Reply:
Thank you for review!
Sounds good. I changed to use logger instead of print. 👉 https://github.com/mocobeta/janome/pull/87/commits/dc737650caab7476d30a9499f3d3ab3455ad99bd

Reply:
Thanks!